tools/setup_minio.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+## see https://github.com/minio/minio/issues/4769 for more examples
+  
+minio_docker_name="mongooseim-minio"
+minio_access_key="AKIAIAOAONIULXQGMOUA"
+minio_secret_key="CG5fGqG0/n6NCPJ10FylpdgRnuV52j8IZvU7BSj8"
+minio_bucket="mybucket"
+
+docker rm -v -f "${minio_docker_name}" || echo "Skip removing previous container"
+
+docker run -d -p 9000:9000 \
+    --name "${minio_docker_name}" \
+    -e "MINIO_ACCESS_KEY=${minio_access_key}" \
+    -e "MINIO_SECRET_KEY=${minio_secret_key}" \
+    minio/minio server /data
+
+mc_cmd="$(cat <<-EOF
+  while ! nc -z minio 9000; do echo 'Wait minio to startup...' && sleep 5; done;
+  mc config host add myminio http://minio:9000 "${minio_access_key}" "${minio_secret_key}" && 
+  mc mb "myminio/${minio_bucket}" &&
+  mc policy set download "myminio/${minio_bucket}"
+  exit 0
+EOF
+)"
```

Comment:
I wonder if `tools/wait_for_service.sh` could be used to wait for the service to startup and only then the config script could be run. I think separation of waiting from configuration may make the configuration part more readable.

Reply:
this will increase the overall waiting time for CI, because we dowload image from dockerhub everytime. also i hope engineers can handle 4 lines of code, minio configuration is much simpler than for other DB containers.

Reply:
As far as I can tell the extra docker container is started for the `wait-for-it.sh` only if we run on MacOS.

Reply:
actually i was talking about `minio/mc` container, now it's started w/o any delay. and then delay is added if needed.

Reply:
Right, that's fair enough.